File: programmers/난이도별/level01.정수_제곱근_판별/JiaLee0707.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long solution(long long n) {
    long long answer = 0;

    if (sqrt(n) == (int)sqrt(n)) {
        // answer = (sqrt(n)+1) * (sqrt(n)+1);
        answer = pow(sqrt(n) + 1, 2);
    }
    else answer = -1;

    return answer;
}

File: programmers/난이도별/level01.3진법_뒤집기/JiaLee0707.js
function ternary(n) {
    let num = [];
    do {
        num.push(n % 3);
        n = parseInt(n/3);
    } while(n > 0);

    return num;
}

function decimal(n) {
    let digit = n.length - 1;
    let num = 0;
    do {
        const themDigit = n.shift();
        num += themDigit * Math.pow(3, digit);
        digit--;
    }while (n.length > 0);
    
    return num;
}

function solution(n) {
    // 1
    // var answer = 0;
    // const ternaryNum = ternary(n);
    // answer = decimal(ternaryNum);
    
    // return answer;
    

    //2
    return parseInt(n.toString(3).split('').reverse().join(''), 3);
}
